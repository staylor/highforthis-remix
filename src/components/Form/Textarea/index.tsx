import React from 'react';
import cn from 'classnames';
import { inputBase } from '@/components/Form/styles';

export default function Textarea({ className, value, ...props }: any) {
  return (
    <textarea {...props} className={cn(inputBase, 'w-full', className)} defaultValue={value} />
  );
}
