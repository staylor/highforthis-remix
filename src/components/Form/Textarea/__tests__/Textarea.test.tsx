import { describe, expect, test, vi } from 'vitest';
import { render, fireEvent } from '@testing-library/react';

import Textarea from '@/components/Form/Textarea';

const TEXT_VALUE = 'Run for the border.';

describe('Textarea', () => {
  test('empty', () => {
    const { container } = render(<Textarea />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('add className', () => {
    const { container } = render(<Textarea className="foo" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('onChange', () => {
    test('adding text', () => {
      const func = vi.fn();
      const { getByRole } = render(<Textarea onChange={func} />);
      const value = TEXT_VALUE;
      fireEvent.change(getByRole('textbox'), { target: { value } });

      expect(func).toHaveBeenCalledWith(value);
    });

    test('removing text', () => {
      const func = vi.fn();
      const value = TEXT_VALUE;
      const { getByRole } = render(<Textarea onChange={func} value={value} />);
      fireEvent.change(getByRole('textbox'), { target: { value: '' } });

      expect(func).toHaveBeenCalledWith('');
    });

    test('bad event', () => {
      const func = vi.fn();
      const value = TEXT_VALUE;
      const { getByRole } = render(<Textarea onChange={func} value={value} />);
      fireEvent.change(getByRole('textbox'), { target: { foo: 'bar', value: null } });

      expect(func).toHaveBeenCalledWith('');
    });

    test('bad type', () => {
      const func = vi.fn();
      const value = TEXT_VALUE;
      const { getByRole } = render(<Textarea onChange={func} value={value} />);
      fireEvent.change(getByRole('textbox'), { target: { value: false } });

      expect(func).toHaveBeenCalledWith('');
    });
  });
});
