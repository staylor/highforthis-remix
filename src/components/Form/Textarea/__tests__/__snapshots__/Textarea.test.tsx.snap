// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea > add className 1`] = `
<textarea
  class="my-1 mr-1 last:mr-0 rounded-md border-gray-300 shadow-sm focus:border-pink focus:ring focus:ring-pink-200 focus:ring-opacity-50 transition-colors w-full foo"
/>
`;

exports[`Textarea > empty 1`] = `
<textarea
  class="my-1 mr-1 last:mr-0 rounded-md border-gray-300 shadow-sm focus:border-pink focus:ring focus:ring-pink-200 focus:ring-opacity-50 transition-colors w-full"
/>
`;
